utils/frame/benchmarking-cli/src/pallet/writer.rs

```diff
@@ -397,38 +447,145 @@ pub(crate) fn add_storage_comments(
 			}
 			let prefix_length = key.len().min(32);
 			let prefix = key[0..prefix_length].to_vec();
-			if identified.contains(&prefix) {
-				// skip adding comments for keys we already identified
-				continue
-			} else {
-				// track newly identified keys
-				identified.insert(prefix.clone());
-			}
-			match storage_info_map.get(&prefix) {
-				Some(key_info) => {
-					let comment = format!(
-						"Storage: {} {} (r:{} w:{})",
-						String::from_utf8(key_info.pallet_name.clone())
-							.expect("encoded from string"),
-						String::from_utf8(key_info.storage_name.clone())
-							.expect("encoded from string"),
-						reads,
-						writes,
-					);
-					comments.push(comment)
+			let is_key_identified = identified_key.contains(key);
+			let is_prefix_identified = identified_prefix.contains(&prefix);
+
+			match (is_key_identified, is_prefix_identified) {
+				// We already did everything, move on...
+				(true, true) => continue,
+				(false, true) => {
+					// track newly identified key
+					identified_key.insert(key.clone());
 				},
-				None => {
-					let comment = format!(
-						"Storage: unknown [0x{}] (r:{} w:{})",
-						HexDisplay::from(key),
-						reads,
-						writes,
-					);
-					comments.push(comment)
+				(false, false) => {
+					// track newly identified key and prefix
+					identified_key.insert(key.clone());
+					identified_prefix.insert(prefix.clone());
 				},
+				// Not possible. If the key is known, the prefix is too.
+				(true, false) => unreachable!(),
+			}
+
+			// For any new prefix, we should write some comment about the number of reads and
+			// writes.
+			if !is_prefix_identified {
+				match storage_info_map.get(&prefix) {
+					Some(key_info) => {
+						let comment = format!(
+							"Storage: {} {} (r:{} w:{})",
+							String::from_utf8(key_info.pallet_name.clone())
+								.expect("encoded from string"),
+							String::from_utf8(key_info.storage_name.clone())
+								.expect("encoded from string"),
+							reads,
+							writes,
+						);
+						comments.push(comment)
+					},
+					None => {
+						let comment = format!(
+							"Storage: unknown [0x{}] (r:{} w:{})",
+							HexDisplay::from(key),
+							reads,
+							writes,
+						);
+						comments.push(comment)
+					},
+				}
+			}
+
+			// For any new key, we should add the PoV impact.
```

Comment:
do these statements have corresponding tests? I can imagine it is hard to write unit tests for these codes, but eventually it would be good to ensure these important assumptions are always held.

Reply:
Yea I added some.